### PR TITLE
feat: don't cascade npm ignored changes to dependencies

### LINKED ISSUE #30: add logic to not release package if only files excluded from npm are changed

### PR DESCRIPTION
**Pull request by kellyselden**
Closes #30

### CODE REVIEW

File: src/build-change-graph.js [hunk 1]
```diff
@@ -116,15 +118,26 @@ async function buildChangeGraph({
...
 
     packagesWithChanges[dag.packageName] = {
       changedFiles: newFiles,
+      changedReleasableFiles,
       dag,
     };
   }
 
-  for (let { dag } of Object.values(packagesWithChanges)) {
+  for (let { dag, changedReleasableFiles } of Object.values(packagesWithChanges)) {
```
Comment (NullVoxPopuli) [inline src/build-change-graph.js:136]: what's `dag`?

Comment (kellyselden) [reply src/build-change-graph.js:136]: directed acyclic graph

File: src/releasable.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    await writeFile(filePath, '');
+  }
+}
+
+// We need to check the changed files rather than
+// what is on disk. A published file could be removed,
+// but it wouldn't look like a new version is needed
+// because it's no longer on disk.
+async function _getChangedReleasableFiles({
```
Comment (NullVoxPopuli) [inline src/releasable.js:56]: thanks for the comment!

File: src/set.js [hunk 3]
```diff
@@ -4,6 +4,31 @@ function union(a, b) {
   return new Set([...a, ...b]);
 }
 
+function intersection(a, b) {
```
Comment (NullVoxPopuli) [inline src/set.js:7]: I feel like these should just exist in Set, Array, etc

File: test/releasable-test.js [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+describe(function() {
+  let tmpPath;
+
+  beforeEach(async function() {
+    tmpPath = await gitInit();
+  });
+
+  describe(getChangedReleasableFiles, function() {
+    it('works with files array', async function() {
+      fixturify.writeSync(tmpPath, {
```
Comment (NullVoxPopuli) [inline test/releasable-test.js:20]: why writeSync when we're in an async function?

**Review by NullVoxPopuli — APPROVED** [review]

File: src/build-change-graph.js (hunk 1 above)
Comment (NullVoxPopuli) [reply src/build-change-graph.js:136]: I understand the need to abbreviate that :D

File: test/releasable-test.js (hunk 4 above)
Comment (kellyselden) [reply test/releasable-test.js:20]: That's all that library offers.

**Pull request merged**